### PR TITLE
refactor `recordSchedulingFailure` function name to more appropriate

### PR DESCRIPTION
**Pull request by kerthcet**
…more rigorous

Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
 Former `recordSchedulingFailure` function does more than just recording the event, refactor the function name then.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ type Scheduler struct {
 	NextPod func() *framework.QueuedPodInfo
 
 	// Error is called if there is an error. It is passed the pod in
-	// question, and the error
+	// queue, and the error
```
Comment (chendave) [inline pkg/scheduler/scheduler.go:83]: I think `in question` is more accurate, you need think that pod is not in the queue now.

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -319,6 +319,7 @@ func (sched *Scheduler) recordSchedulingFailure(fwk framework.Framework, podInfo
 		Reason:  reason,
 		Message: err.Error(),
 	}, nominatedNode); err != nil {
+		sched.SchedulingQueue.DeleteNominatedPodIfExists(podInfo.Pod)
```
Comment (chendave) [inline pkg/scheduler/scheduler.go:322]: Could you pls add a comments to explain in which case the pod is still in the queue? This doesn't seems like a normal case.

Comment (kerthcet) [reply pkg/scheduler/scheduler.go:322]: It's just a rollback operator of `AddNominatedPod`. 
When we preempted successfully, we'll add the `nominatedNode` to the SchedulingQueue by calling `AddNominatedPod()`, after then we will update pod's status, but if some error happens here, the pod's status would be dis-match with the cache. 
Although in the next preemption cycle, we may remove the cache by calling the `AddNominatedPod`,  but I think add a revert operator here is better.

File: pkg/scheduler/scheduler.go (hunk 1 above)
Comment (kerthcet) [reply pkg/scheduler/scheduler.go:83]: thanks. done

**Comment by ahg-g** [comment]
@Huang-Wei is this related to https://github.com/kubernetes/kubernetes/pull/106816?

**Comment by Huang-Wei** [comment]
> @Huang-Wei is this related to #106816?

Not quite. This PR _tries_ to add a reverse operation for AddNomiantedNode when updatePod() failed, but I'm opposed to this change as the semantics of AddNomiantedNode() is complex that its name indicates, like it sometimes updates the internal nominator cache, enforce a DeleleNominatedNode here can cause unexpected behavior.

**Comment by kerthcet** [comment]
close then.
/close

**Comment by kerthcet** [comment]
reopen as code cleanup.
/reopen

**Comment by kerthcet** [comment]
/retitle refactor `recordSchedulingFailure` function name to more appropriate

**Pull request merged**